### PR TITLE
[SOW MS3] Enable llvm for tensorexpr, test_openmp

### PR DESCRIPTION
**Pull request by arindamroy-eng**
The following 12 tests are being re-enabled for ROCM:

test_openmp test_n_threads (__main__.TestOpenMP_ParallelFor)
test_openmp test_one_thread (__main__.TestOpenMP_ParallelFor)
And
test_tensorexpr_pybind test_alloc_in_loop (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_shape_prop (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_shape_prop_module (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_custom_lowering (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_expand (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_permute (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_scalar_inputs (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_t (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_tensor_inputs (main.TestTensorExprPyBind)
test_tensorexpr_pybind test_kernel_with_transpose (main.TestTensorExprPyBind)

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@arindamroy-eng CI is still busted. :( Would you mind posting a snippet of the passing unit tests from your local run logs? Please mention the docker image you used.

**Comment by arindamroy-eng** [comment]
@jithunnair-amd I posting the logs:
As for the exact container, as I do not have the exact build info right now, but seems like I tested on 11_ubuntu18.04_py3.7_pytorch_rocm5.1_internal_testing_08a7cc9
and also 5.2 (CI Image).
 
test_alloc_in_loop (__main__.TestTensorExprPyBind) ... ok
test_call_raw (__main__.TestTensorExprPyBind) ... ok
test_dtype_error (__main__.TestTensorExprPyBind) ... ok
test_dynamic_shape (__main__.TestTensorExprPyBind) ... ok
test_dynamic_shape_2d (__main__.TestTensorExprPyBind) ... ok
test_external_calls (__main__.TestTensorExprPyBind) ... ok
test_kernel_shape_prop (__main__.TestTensorExprPyBind) ... ok
test_kernel_shape_prop_module (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_custom_lowering (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_expand (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_permute (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_scalar_inputs (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_t (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_tensor_inputs (__main__.TestTensorExprPyBind) ... ok
test_kernel_with_transpose (__main__.TestTensorExprPyBind) ... ok
test_simple_sum (__main__.TestTensorExprPyBind) ... ok

test_n_threads (__main__.TestOpenMP_ParallelFor)
Make sure there is no memory leak with many threads ... ok
test_one_thread (__main__.TestOpenMP_ParallelFor)
Make sure there is no memory leak with one thread: issue gh-32284 ... ok

**Comment by jithunnair-amd** [comment]
@arindamroy-eng how does this PR enable the test_tensorexpr_pybind tests? Can you please point to the exact line(s) or file(s)?

File: .jenkins/pytorch/test.sh [hunk 1]
```diff
@@ -303,10 +303,12 @@ test_libtorch() {
 }
 
 test_aot_compilation() {
-  echo "Testing Ahead of Time compilation"
```
Comment (jithunnair-amd) [inline .jenkins/pytorch/test.sh:306]: How is this change related to the PR at all?

**Comment by arindamroy-eng** [comment]
Closing this PR as new PR has been submitted.

File: .jenkins/pytorch/test.sh (hunk 1 above)
Comment (jithunnair-amd) [reply .jenkins/pytorch/test.sh:306]: Removed in resubmitted PR #1064
